### ISSUE TITLE
Data list: add static header 

### ISSUE THREAD

**Issue by jenny-s51** [patternfly/patternfly-react#5741]
**Describe the issue. What is the expected and unexpected behavior?**
Currently the data list component does not support a header like the table component does. If using a separate `DataListItem` as a header, the draggable variant of the data table causes the data cell move around if the user selects an item to drag. In this use case, the first cell should always remain static/unmovable when dragging around items. See below.

![ezgif com-gif-maker](https://user-images.githubusercontent.com/32821331/117015256-867cf800-acbf-11eb-8f68-09ce72123bd5.gif)

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**
feature request

**What is your product and what release version are you targeting?**
Keycloak - new admin console

**Comment by tlabaj** [comment]
cc @mcarrano

**Comment by mcarrano** [comment]
@jenny-s51 Data lists are not intended to have static headers. If you want fixed columns in a list, then the Table is the preferred approach. Is there a reason that a data list is being used here instead of a table?

**Comment by jenny-s51** [comment]
@mcarrano yes, the Table does not support draggable rows

**Comment by mcarrano** [comment]
In that case, I would ideally prefer to add draggable rows to the Table rather than create what is essentially a faux-table. @tlabaj @nicolethoen are there any technical reasons why we couldn't do that?

**Comment by jenny-s51** [comment]
That's a great idea @mcarrano - I've renamed this issue accordingly. Do you think it would be possible to get it into the upcoming sprint @tlabaj @nicolethoen ?

**Comment by redallen** [comment]
We cannot add draggable functionality to the legacy table without technical difficulties (how can we get `ref`s to the `tr` elements? How should dragging + filtering at the same time work?). However, we might be able to copy the vanilla JS used for dragging in `Datalist` and add it to a `ComposableTable` demo.

**Comment by mcarrano** [comment]
I think that it would be OK to do this with a composable table if that approach makes sense. @jenny-s51 can you take a look at the composable table implementation and see if that's something that would work for you? If not, then we may need to find a different solution.

**Comment by jenny-s51** [comment]
Yes that sounds like a great solution. Thank you for looking into this @redallen @mcarrano

**Comment by mcarrano** [comment]
@tlabaj @mcoker what do we need then to take that approach? Do we need to start with a core issue for the draggable row?

**Comment by mcoker** [comment]
@mcarrano yep, we'll need to add the draggable/ghost row styling currently in data list to the table.

**Comment by mcarrano** [comment]
In that case I will transfer this to Core so we can start the work there and get it on the roadmap. @jenny-s51 do you know if there are any specific release targets/timeframes for this?
